src/interface/src/app/plan/create-scenarios/create-scenarios.component.ts

```diff
@@ -64,14 +71,19 @@ export class CreateScenariosComponent implements OnInit {
   constructor(
     private fb: FormBuilder,
     private planStateService: PlanStateService,
+    private scenarioService: ScenarioService,
     private router: Router,
     private matSnackBar: MatSnackBar,
     private featureService: FeatureService
   ) {}
 
   createForms() {
     this.forms = this.fb.group({
-      scenarioName: [null, Validators.required],
+      scenarioName: new FormControl('', [
+        Validators.required,
+        (control: AbstractControl) =>
+          scenarioNameMustBeNew(control, this.existingScenarioNames),
```

Comment:
thoughts on this? this seemed to be the only way to copy the scenario names to the validator